awslabs/aws-athena-query-federation
# Add Redis Integ Tests

*Description of changes:*
Adding Integration Tests for the Redis Connector. 

Two Redis instances (standalone, and clustered) will be deployed. Just like in the DocDB integ tests, an extra helper lambda will be deployed to populate the test data into the Redis instances.
There are 15 tests in total:
```
[INFO] Tests run: 15, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1,196.281 s - in com.amazonaws.athena.connectors.redis.integ.RedisIntegTest
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 15, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 19:58 min
[INFO] Finished at: 2021-08-18T01:02:51-04:00
```
3 Tests for checking DB, Tables, Schema
6 Tests for each Redis Instance (standalone, and clustered with ssl+password)
- Select Keys provided by Prefix with Hash Value
- Select Keys provided by Prefix with Zset Value
- Select Keys provided by Prefix with Literal Value
- Select Keys provided by Zset with Hash Value
- Select Keys provided by Zset with Zset Value
- Select Keys provided by Zset with Literal Value

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
